pkg/provider/azure_loadbalancer.go

```diff
@@ -262,6 +262,10 @@ func (az *Cloud) shouldChangeLoadBalancer(service *v1.Service, currLBName, clust
 	// if the current LB is what we want, keep it
 	lbName := strings.TrimSuffix(currLBName, consts.InternalLoadBalancerNameSuffix)
 	if strings.EqualFold(lbName, vmSetName) {
+		if strings.EqualFold(az.VMSet.GetPrimaryVMSetName(), vmSetName) {
```

Comment:
should we check clusterName here as well? If clusterName==vmSetName, then no need to change the slb.

Reply:
it's possible on unmanaged clusters.

Reply:
updated